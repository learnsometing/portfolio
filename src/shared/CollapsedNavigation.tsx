import React, { useState } from 'react';
import styled from 'styled-components';
import Hamburger from 'hamburger-react';

import Typography from '@material-ui/core/Typography';
import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import Divider from '@material-ui/core/Divider';
import { NavLink } from './Navigation';

const Toggle = styled.div`
  width: 60px;
  height: 60px;
  position: fixed;
  bottom: 0;
  right: 0;
  z-index: 1060;
  margin-bottom: 1rem;
  margin-right: 1rem;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 50% !important;
  background-color: #f4d35e !important;
  box-shadow: 0px 3px 5px rgba(0%, 0%, 0%, 0.5);

  @media only screen and (min-width: 768px) {
    display: none;
  }
`;

const Menu = styled(List)`
  height: 100vh;
  width: 100vw;
`;

const MenuItem = styled(ListItem).attrs({
  button: undefined,
})`
  height: 25%;
  margin: 0;
`;

const MenuLink = styled(NavLink)`
  display: flex;
  width: 100%;
  height: 100%;
  padding: 0;
  justify-content: center;
  align-items: center;
`;

const CollapsedNavigation: React.FC = () => {
  const [isDrawerOpen, setIsDrawerOpen] = useState(false);

  const toggleDrawer = (): void => {
    setIsDrawerOpen(!isDrawerOpen);
  };

  return (
    <>
      <Toggle>
        <Hamburger
          color="#0d3b66"
          toggled={isDrawerOpen}
          toggle={toggleDrawer}
        />
      </Toggle>
      <Drawer anchor={'top'} open={isDrawerOpen}>
        <Menu>
          <MenuItem>
            <MenuLink to="/">
              <Typography variant={'body1'} component="span">
                Home
              </Typography>
            </MenuLink>
          </MenuItem>
          <Divider component={'li'} />
          <MenuItem>
            <MenuLink to="/portfolio">
              <Typography variant={'body1'} component="span">
                Portfolio
              </Typography>
            </MenuLink>
          </MenuItem>
          <Divider component={'li'} />
          <MenuItem>
            <MenuLink to="/about">
              <Typography variant={'body1'} component="span">
                About
              </Typography>
            </MenuLink>
          </MenuItem>
          <Divider component={'li'} />

          <MenuItem>
            <MenuLink to="/contact">
              <Typography variant={'body1'} component="span">
                Contact
              </Typography>
            </MenuLink>
          </MenuItem>

          <Toggle>
            <Hamburger
              color="#0d3b66"
              toggled={isDrawerOpen}
              toggle={toggleDrawer}
            />
          </Toggle>
        </Menu>
      </Drawer>
    </>
  );
};

export default CollapsedNavigation;
