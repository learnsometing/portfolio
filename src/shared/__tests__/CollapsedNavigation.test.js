import React from 'react';
import { create, act } from 'react-test-renderer';
import { fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import CollapsedNavigation from '../CollapsedNavigation';

describe('CollapsedNavigation', () => {
  let tree;
  beforeEach(() =>
    act(() => {
      tree = create(<CollapsedNavigation />);
    })
  );
  it('should match the snapshot', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
