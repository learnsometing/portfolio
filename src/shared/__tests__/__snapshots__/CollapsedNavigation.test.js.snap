// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsedNavigation should match the snapshot 1`] = `
.c0 {
  width: 60px;
  height: 60px;
  position: fixed;
  bottom: 0;
  right: 0;
  z-index: 1060;
  margin-bottom: 1rem;
  margin-right: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50% !important;
  background-color: #f4d35e !important;
  box-shadow: 0px 3px 5px rgba(0%,0%,0%,0.5);
}

@media only screen and (min-width:768px) {
  .c0 {
    display: none;
  }
}

<div
  className="c0"
>
  <div
    className="hamburger-react"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="button"
    style={
      Object {
        "cursor": "pointer",
        "height": "48px",
        "outline": "none",
        "position": "relative",
        "transform": "none",
        "transition": "0.4s cubic-bezier(0, 0, 0, 1)",
        "userSelect": "none",
        "width": "48px",
      }
    }
    tabIndex={0}
  >
    <div
      style={
        Object {
          "background": "#0d3b66",
          "height": "3px",
          "left": "8px",
          "position": "absolute",
          "top": "13px",
          "transform": "none",
          "transition": "0.4s cubic-bezier(0, 0, 0, 1)",
          "width": "32px",
        }
      }
    />
    <div
      style={
        Object {
          "background": "#0d3b66",
          "height": "3px",
          "left": "8px",
          "position": "absolute",
          "top": "23px",
          "transform": "none",
          "transition": "0.4s cubic-bezier(0, 0, 0, 1)",
          "width": "32px",
        }
      }
    />
    <div
      style={
        Object {
          "background": "#0d3b66",
          "height": "3px",
          "left": "8px",
          "position": "absolute",
          "top": "33px",
          "transform": "none",
          "transition": "0.4s cubic-bezier(0, 0, 0, 1)",
          "width": "32px",
        }
      }
    />
  </div>
</div>
`;
